Foundation/NSMeasurement.swift

```diff
@@ -23,21 +23,95 @@ open class NSMeasurement : NSObject, NSCopying, NSSecureCoding {
         self.unit = unit
     }
     
-    open func canBeConverted(to unit: Unit) -> Bool { NSUnimplemented() }
+    open func canBeConverted(to otherUnit: Unit) -> Bool {
+        #if DEPLOYMENT_RUNTIME_OBJC //|| os(Linux)
+            return otherUnit.isKind(of: type(of: unit))
+        #else
+            // just check conversion
+            if let dimension = unit as? Dimension,
+               let otherDimension = otherUnit as? Dimension {
+                return true
+            } else {
+                return unit.isEqual(otherUnit)
+            }
+            re
+        #endif
+    }
     
-    open func converting(to unit: Unit) -> Measurement<Unit> { NSUnimplemented() }
+    open func converting(to otherUnit: Unit) -> Measurement<Unit> {
+        if canBeConverted(to: otherUnit) {
+            if unit.isEqual(otherUnit) {
+                return Measurement(value: doubleValue, unit: otherUnit)
+            } else {
+                guard let dimension = unit as? Dimension,
+                    let otherDimension = otherUnit as? Dimension else {
+                        fatalError("Cannot convert differing units that are non-dimensional! lhs: \(type(of: unit)) rhs: \(type(of: otherUnit))")
+                }
+                let valueInTermsOfBase = dimension.converter.baseUnitValue(fromValue: doubleValue)
+                if otherDimension.isEqual(type(of: dimension).baseUnit()) {
+                    return Measurement(value: valueInTermsOfBase, unit: otherDimension)
+                } else {
+                    let otherValueFromTermsOfBase = otherDimension.converter.value(fromBaseUnitValue: valueInTermsOfBase)
+                    return Measurement(value: otherValueFromTermsOfBase, unit: otherDimension)
+                }
+            }
+        } else {
+            fatalError("Cannot convert measurements of differing unit types! self: \(type(of: unit)) unit: \(type(of: otherUnit))")
+        }
+    }
     
-    open func adding(_ measurement: Measurement<Unit>) -> Measurement<Unit> { NSUnimplemented() }
+    open func adding(_ rhs: Measurement<Unit>) -> Measurement<Unit> {
+        if self.unit.isEqual(rhs.unit) {
+            return Measurement(value: self.doubleValue + rhs.value, unit: self.unit)
+        } else {
+            guard let dimension = unit as? Dimension,
+                    let otherDimension = otherUnit as? Dimension else {
```

Comment:
Do you mean `rhs` instead of `otherUnit` ? (same on line 82)

Reply:
Yes, indeed.